panoptes_cli/commands/subject.py

```diff
@@ -51,6 +64,282 @@ def ls(subject_set_id, quiet, subject_ids):
+    uploaded, or filename columns can be specified with --file-column.
+
+    If you are hosting your media yourself, you can put the URLs in the
+    manifest and specify the column number(s):
+
+    $ panoptes subject upload-subjects-attached-media -r 1  manifest.csv
+
+    $ panoptes subject upload-subjects-attached-media -r 1 -r 2  manifest.csv
+
+    Any local files will still be detected and uploaded.
+    """
+    if (
+        len(manifest_files) > 1
+        and any(map(lambda m: m.endswith('.yaml'), manifest_files))
+    ):
+        click.echo(
+            'Error: YAML manifests must be processed one at a time.',
+            err=True,
+        )
+        return -1
+    elif manifest_files[0].endswith('.yaml'):
+        with open(manifest_files[0], 'r') as yaml_manifest:
+            upload_state = yaml.load(yaml_manifest, Loader=yaml.FullLoader)
+        if upload_state['state_version'] > CURRENT_STATE_VERSION:
+            click.echo(
+                'Error: {} was generated by a newer version of the Panoptes '
+                'CLI and is not compatible with this version.'.format(
+                    manifest_files[0],
+                ),
+                err=True,
+            )
+            return -1
+        resumed_upload = True
+    else:
+        upload_state = {
+            'state_version': CURRENT_STATE_VERSION,
+            'manifest_files': manifest_files,
+            'allow_missing': allow_missing,
+            'remote_location': remote_location,
+            'mime_type': mime_type,
+            'file_column': file_column,
+            'waiting_to_upload': []
+        }
+        resumed_upload = False
+
+    remote_location_count = len(upload_state['remote_location'])
+    mime_type_count = len(upload_state['mime_type'])
+    if remote_location_count > 1 and mime_type_count == 1:
+        upload_state['mime_type'] = (
+            upload_state['mime_type'] * remote_location_count
+        )
+    elif remote_location_count > 0 and mime_type_count != remote_location_count:
+        click.echo(
+            'Error: The number of MIME types given must be either 1 or equal '
+            'to the number of remote locations.',
+            err=True,
+        )
+        return -1
+
+    if not resumed_upload:
+        subject_rows = []
+        subjects_by_id = {}
+        for manifest_file in upload_state['manifest_files']:
+            with open(manifest_file) as manifest_f:
+                file_root = os.path.dirname(manifest_file)
+                r = csv.reader(manifest_f, skipinitialspace=True)
+                headers = next(r)
+                for row_number, row in enumerate(r, start=1):
+                    metadata = dict(zip(headers, row))
+                    files = []
+                    if not metadata.get('subject_id'):
+                        click.echo(f'Missing subject_id in row:{row_number}', err=True)
+                        return -1
+                    subject_id = metadata['subject_id']
```

Comment:
Remove the subject_id key from media metadata (that would be duplicative).
```suggestion
                    subject_id = metadata.pop('subject_id')
```